djautotask/api.py

```diff
@@ -498,10 +497,14 @@ def _fetch_resource(endpoint_url, request_retry_counter=None,
                 logger.warning(msg)
                 if request_retry_counter['count'] <= self.MAX_401_ATTEMPTS:
                     cached_url = get_cached_url(AT_URL_KEY)
-                    if cached_url != get_api_connection_url(force_fetch=True):
-                        logger.info('Zone information has been changed, '
-                                    'so this request will be retried.')
-                        raise AutotaskAPIError(response.content)
+                    try:
+                        if cached_url != \
+                                get_api_connection_url(force_fetch=True):
+                            logger.info('Zone information has been changed, '
+                                        'so this request will be retried.')
+                            raise AutotaskAPIError(response.content)
+                    except AutotaskAPIError as e:
+                        raise AutotaskAPIError('{}'.format(e))
```

Comment:
This area originally raised an `AutotaskAPIError` and anything calling it should expect to have to handle that error, so when `get_api_connection_url` is used here there isn't a problem. If the condition is true it raises the exception, or if there is a problem in the method call it raises the same exception. So no changes need to be made here, and again you are just re-raising the caught exception so any unhandled exceptions would still unhandled.

The problem is on line 301:

```
      if not server_url:
            server_url = get_api_connection_url()
```
Where a developer might not expect that error when instantiating the class. So you need to add handling to the code that is instantiating the api class, to catch and handle the `AutotaskAPIError`. This would be easy, but it appears sentry has lost the stack trace, so we might have to track it down. If I had to guess right now (I'll look to confirm later), its in the sync code where it instantiates the API class to perform the syncs.